src/libs/actions/Policy.ts

```diff
@@ -274,33 +288,37 @@ function buildAnnounceRoomMembersOnyxData(policyID, accountIDs) {
         return announceRoomMembers;
     }
 
-    announceRoomMembers.onyxOptimisticData.push({
-        onyxMethod: Onyx.METHOD.MERGE,
-        key: `${ONYXKEYS.COLLECTION.REPORT}${announceReport.reportID}`,
-        value: {
-            participantAccountIDs: [...announceReport.participantAccountIDs, ...accountIDs],
-        },
-    });
+    if (announceReport?.participantAccountIDs) {
+        announceRoomMembers.onyxOptimisticData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${announceReport?.reportID}`,
+            value: {
+                participantAccountIDs: [...announceReport.participantAccountIDs, ...accountIDs],
+            },
+        });
+    }
 
     announceRoomMembers.onyxFailureData.push({
         onyxMethod: Onyx.METHOD.MERGE,
-        key: `${ONYXKEYS.COLLECTION.REPORT}${announceReport.reportID}`,
+        key: `${ONYXKEYS.COLLECTION.REPORT}${announceReport?.reportID}`,
         value: {
-            participantAccountIDs: announceReport.participantAccountIDs,
+            participantAccountIDs: announceReport?.participantAccountIDs,
         },
     });
     return announceRoomMembers;
 }
 
+type OptimisticAnnounceRoomMembers = {
+    onyxOptimisticData: OnyxUpdate[];
+    onyxFailureData: OnyxUpdate[];
+};
+
```

Comment:
NAB: Maybe let's move this to top of the file? Taking into consideration how we did it in ReportUtils with Optimistic types